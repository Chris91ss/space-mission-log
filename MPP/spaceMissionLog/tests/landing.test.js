import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import LandingPage from '../client/pages/index';
import '@testing-library/jest-dom';

describe('LandingPage', () => {
  it('renders the video element with correct attributes (excluding muted)', async () => {
    const { container } = render(<LandingPage />);
    const video = await waitFor(() =>
      container.querySelector('[data-testid="landing-video"]')
    );
    expect(video).toBeInTheDocument();
    expect(video).toHaveAttribute('preload', 'auto');
    expect(video).toHaveAttribute('loop');
    expect(video).toHaveAttribute('playsInline');

    const source = video.querySelector('source');
    expect(source).toBeInTheDocument();
    expect(source).toHaveAttribute('src', '/rocket_motion.mp4');
    expect(source).toHaveAttribute('type', 'video/mp4');
  });

  it('renders the title, subtitle, and enter button', () => {
    render(<LandingPage />);
    expect(screen.getByText('Space Mission Log System')).toBeInTheDocument();
    expect(screen.getByText('Track every mission detail in one place')).toBeInTheDocument();
    const enterButton = screen.getByRole('button', { name: /Enter/i });
    expect(enterButton).toBeInTheDocument();
  });

  it('matches the snapshot', async () => {
    const { asFragment } = render(<LandingPage />);
    await waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
