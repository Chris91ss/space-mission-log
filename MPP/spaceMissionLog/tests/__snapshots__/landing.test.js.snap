// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <nav
      class="navbar"
    >
      <div
        class="logo"
      >
        <a
          href="/"
        >
          Space Mission Log
        </a>
      </div>
      <ul
        class="navLinks"
      >
        <li>
          <a
            href="/missions"
          >
            Missions
          </a>
        </li>
        <li>
          <a
            href="/missions/create"
          >
            Add Mission
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="mainContent"
    >
      <div
        class="landingContainer"
      >
        <div
          class="videoBackground"
        >
          <video
            autoplay=""
            class="video"
            data-testid="landing-video"
            loop=""
            playsinline=""
            preload="auto"
          >
            <source
              src="/rocket_motion.mp4"
              type="video/mp4"
            />
            Your browser does not support the video tag.
          </video>
        </div>
        <div
          class="overlay"
        >
          <h1
            class="title"
          >
            Space Mission Log System
          </h1>
          <p
            class="subtitle"
          >
            Track every mission detail in one place
          </p>
          <a
            href="/missions"
          >
            <button
              class="enterButton"
            >
              Enter
            </button>
          </a>
        </div>
      </div>
    </main>
  </div>
</DocumentFragment>
`;
